.github/workflows/build.yml

```diff
@@ -0,0 +1,75 @@
+name: Test and Build Workflow
+# This workflow is triggered on pull requests to main branch
+on:
+  pull_request:
```

Comment:
You should probably have CI on all branches.

Reply:
Yes we should. But right now we've multiple dev branches and it seems wasteful to run CI in all branches. 
We'll just tracking main for now.